### PR TITLE
Hosting: Copy updates and add Clipboard Fields for SFTP

### PR DESCRIPTION
**Pull request by gwwar**
This PR addresses various feedback concerns noted in https://github.com/Automattic/wp-calypso/issues/37979

## Copy Updates

Copy within the component is very likely to get stale, so the goal here was to be succinct as possible and link to the correct support documentation.

### Subheading when SFTP is enabled

| Before  | After |
| ------------- | ------------- |
| Screen Shot 2019-11-26 at 5 12 15 PM | Screen Shot 2019-11-27 at 2 12 42 PM |

### SFTP Credentials:

| Before  | After |
| ------------- | ------------- |
| Screen Shot 2019-11-26 at 4 48 09 PM| Screen Shot 2019-11-27 at 2 14 32 PM |

### Database Access:

| Before  | After |
| ------------- | ------------- |
| Screen Shot 2019-11-26 at 4 51 45 PM | Screen Shot 2019-11-27 at 2 14 41 PM |

### SFTP Button
Enable vs Create vs Generate?

| Before | After |
| ------------- | ------------- |
| Screen Shot 2019-11-26 at 5 18 34 PM| Screen Shot 2019-11-26 at 5 18 17 PM |

## Update SFTP table to form fields

This is what this currently looks like:

| Before | After |
| ------------- | ------------- |
| Screen Shot 2019-11-26 at 5 04 57 PM | Screen Shot 2019-11-27 at 4 00 36 PMScreen Shot 2019-11-27 at 4 01 05 PM|

Notice that the subheader for the SFTP card changes depending on if an SFTP user exists:
Screen Shot 2019-11-27 at 4 01 20 PM

### CODE REVIEW

File: client/my-sites/hosting/main.js [hunk 1]
```diff
@@ -43,7 +43,13 @@ const Hosting = ( { translate, isDisabled, canViewAtomicHosting, siteId } ) => {
 			<SidebarNavigation />
 			<FormattedHeader
 				headerText={ translate( 'Hosting Configuration' ) }
-				subHeaderText={ translate( 'Access and manage more advanced settings of your website.' ) }
+				subHeaderText={
+					sftpPhpMyAdminFeaturesEnabled
+						? translate(
+								'Access your website, database, and more advanced settings of your website.'
```
Comment (a8ci18n) [inline client/my-sites/hosting/main.js]: :warning: This change will be queued for retranslation. We'll display the English text until then.

File: client/my-sites/hosting/sftp-card/index.js [hunk 2]
```diff
@@ -183,7 +176,7 @@ const SftpCard = ( {
 						) }
 					</p>
 					<Button onClick={ createUser } primary>
-						{ translate( 'Enable SFTP Credentials' ) }
+						{ translate( 'Create SFTP Credentials' ) }
```
Comment (a8ci18n) [inline client/my-sites/hosting/sftp-card/index.js]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( 'Enable SFTP Credentials' )` *ES Score: 7*

File: client/my-sites/hosting/sftp-card/index.js [hunk 3]
```diff
@@ -143,27 +145,18 @@ const SftpCard = ( {
...
 					<Accordion title={ translate( 'What is SFTP?' ) }>
 						{ translate(
-							"{{p}}It's a way for you to access the files and folders on a website via a client program such as {{a}}Filezilla{{/a}} on your local computer.{{/p}}" +
-								'{{p}}SFTP stands for Secure File Transfer Protocol (or SSH File Transfer Protocol). It was designed as an extension of the SSH (Secure SHell) protocol. The “secure” part is because it is run over a secure channel, in this case SSH.{{/p}}' +
-								'{{p}}SFTP is not to be confused with File Transfer Protocol (FTP) which is similar but not secure. Thanks to the security provided by SFTP you can rest easy knowing that your files and your site are safe and sound.{{/p}}' +
-								'{{p}}A variety of custom plugins and themes may ask you to create specific folders or add files via SFTP however, generally speaking, SFTP is not required for your site to function.{{/p}}',
+							"It's a way for you to access the files and folders on a website via a client program such as {{a}}Filezilla{{/a}} on your local computer. " +
+								'For more information see {{supportLink}}SFTP and WordPress.com{{/supportLink}} ',
```
Comment (mmtr) [inline client/my-sites/hosting/sftp-card/index.js]: The support page title is "SFTP on WordPress.com": https://en.support.wordpress.com/sftp/

```suggestion
								'For more information see {{supportLink}}SFTP on WordPress.com{{/supportLink}} ',
```

Comment (gwwar) [reply client/my-sites/hosting/sftp-card/index.js]: Thanks! wow the GitHub suggestion feature is fancy

**Comment by carolynannewells** [comment]
You have tightened up these descriptions really well! I just have a few copy suggestions for you:


**Subheading when SFTP is enabled**

Access your website’s database and more advanced settings.


**SFTP credentials** 

It’s a secure way for you to access your website files on your local computer via a client program such Filezilla. 


**Database Access**


It is a free open source software tool…_(take out 'and')_

Managing a database can be tricky and it’s not necessary for your site to function.

**Comment by gwwar** [comment]
Note that the updated SFTP form fields are using "Form Text Input With Action" https://wpcalypso.wordpress.com/devdocs/design/form-fields

**Comment by gwwar** [comment]
Thanks @carolynannewells, I've applied changes in https://github.com/Automattic/wp-calypso/pull/37991/commits/da4dc3c1e95f21993d1e862fc4ddc9b3f1d65f78

@olaolusoga I'll tackle SFTP field styling updates next.

**Comment by olaolusoga** [comment]
>maybe if we aligned left whitespace for the largest breakpoint, with left text align for the table headers?

What @mmtr shared. We should match screenshot.

**Note on the sub-header copy for the SFTP card:** The sub-header copy should be different before, and after creating credentials. 
The former asks folks to create credentials "Access and edit your website’s files directly by creating SFTP credentials, and using an FTP client.". The former also provides quick general, evergreen info for context and points folks to support docs for more info

Screenshot before creating credentials:
Screen Shot 2019-11-27 at 5 23 23 PM


The latter tells folks to use their credentials and can link out to documentation for more context if needed "Use the credentials below to access and edit your website files using an FTP client. [Learn more about SFTP and WordPress.com.]". At this point, the accordion shouldn't be there, since the focus is on the credentials. 

The cases where there may be multiple admins on a site, or sites sometimes change hands, the link to learn more in the subheader should be enough for guidance. The accordion does not answer all questions; it was met for quick guidance before the credentials are created. Once created, the credentials are the focus, and we should point folks to the support docs to learn more. Also in both instances, assuming admins and folks talk to each other about SFTP when working on their sites.

Screenshot of after credential creation:
Screen Shot 2019-11-27 at 5 20 15 PM

File: client/my-sites/hosting/sftp-card/index.js [hunk 4]
```diff
@@ -136,34 +135,42 @@ const SftpCard = ( {
...
-								components: {
-									p: <p />,
-									a: <a href={ FILEZILLA_URL } />,
-								},
-							}
-						) }
-					</Accordion>
-					<Accordion title={ translate( 'How to use SFTP to access your site' ) }>
-						{ translate(
-							'{{p}}Click on "Enable SFTP Credentials". Once you’ve enabled SFTP, you’ll see your SFTP URL, Port Number, Username and Password. Keep these credentials safe!{{/p}}' +
```
Comment (a8ci18n) [inline client/my-sites/hosting/sftp-card/index.js]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( 'Enable SFTP Credentials' )` *ES Score: 7*

**Comment by gwwar** [comment]
Thanks Ola, this one is ready for another look! All functions in the SFTP card probably need another manual test from a +1 since I had to refactor the entire table.

File: client/my-sites/hosting/sftp-card/index.js [hunk 5]
```diff
@@ -92,21 +92,23 @@ const SftpCard = ( {
...
 		}
 
 		if ( password ) {
 			return (
 				<>
-					<span className="sftp-card__hidden-overflow">{ password }</span>
-					<ClipboardButton text={ password } onCopy={ () => setIsCopied( 'password' ) } compact>
-						{ passwordIsCopied ? translate( 'Copied!' ) : translate( 'Copy password' ) }
-					</ClipboardButton>
+					<div className="sftp-card__copy-field">
```
Comment (glendaviesnz) [inline client/my-sites/hosting/sftp-card/index.js]: 👍 nice simple way to implement this design!

File: client/my-sites/hosting/sftp-card/index.js [hunk 6]
```diff
@@ -92,21 +92,23 @@ const SftpCard = ( {
...
+						<FormTextInput className="sftp-card__copy-input" value={ password } onChange={ noop } />
+						<ClipboardButton className="sftp-card__copy-button" text={ password } compact>
+							{ translate( 'Copy' ) }
```
Comment (mmtr) [inline client/my-sites/hosting/sftp-card/index.js]: Copy has more than 1 meaning, it'd be helpful to provide context for translators

File: client/my-sites/hosting/sftp-card/index.js [hunk 7]
```diff
@@ -136,34 +135,42 @@ const SftpCard = ( {
...
+												icon
+												target="_blank"
+												href={ localizeUrl( 'https://en.support.wordpress.com/sftp/' ) }
+											/>
+										),
+									},
+								}
+						  )
+						: translate(
+								"Access and edit your website's files directly by creating SFTP credentials and using an SFTP client."
```
Comment (a8ci18n) [inline client/my-sites/hosting/sftp-card/index.js]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Access and edit your website's files directly by using an SFTP client.' )` *ES Score: 7*

**Comment by mmtr** [comment]
I don't think the failed `wp-desktop` build is related to changes on this PR.

**Comment by glendaviesnz** [comment]
This still tested well for me after @mmtr's changes so going to go ahead and merge.

File: client/my-sites/hosting/sftp-card/index.js [hunk 8]
```diff
@@ -115,13 +117,10 @@ const SftpCard = ( {
...
 			<>
-				<span>{ translate( 'You must reset your password to view it.' ) }</span>
-				<Button
-					onClick={ resetPassword }
-					disabled={ isPasswordLoading }
-					busy={ isPasswordLoading }
-					compact
-				>
+				<p className="sftp-card__password-explainer">
+					{ translate( 'For security reasons, your password needs a reset to view' ) }
```
Comment (akirk) [inline client/my-sites/hosting/sftp-card/index.js]: This sentence seems cut off, could you improve the copy?

Comment (mmtr) [reply client/my-sites/hosting/sftp-card/index.js]: https://github.com/Automattic/wp-calypso/pull/38243

**Pull request merged**